### PR TITLE
[Fix] timer retain issue in ConversationCell and GiphySearchViewController

### PR DESCRIPTION
**Pull request by billypchan**
### Issues

In ConversationCell and GiphySearchViewController, Timers were created with  Timer.scheduledTimer(timeInterval: target:self  selector:  userInfo: , repeats: ) which cause self is still retained by the timer and can not be freed.

### Causes
Timer.scheduledTimer(timeInterval: target: selector:  userInfo: , repeats: ) makes a strong reference to target. 

### Solutions

Rewrite Timer.scheduledTimer related call with scheduledTimer(withTimeInterval interval: TimeInterval, repeats: Bool, block: @escaping (Timer) -> Swift.Void) and capture self weakly in the block. Create a scheduledTimer with block method for iOS9-.

## Notes

UIApplication (runDuration), NetworkStatusViewController and VoiceChannelViewController are still using Timer.scheduledTimer(timeInterval: target:  selector:  userInfo: , repeats: ). Will check they are necessary or not in next PR.

### CODE REVIEW

**Comment by billypchan** [comment]
This fix is already done in https://github.com/wireapp/wire-ios/pull/1558

File: Wire-iOS/Sources/Helpers/Timer+allVersionCompatibleScheduledTimer.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    static func allVersionCompatibleScheduledTimer(withTimeInterval: TimeInterval, repeats: Bool, block: @escaping (Timer) -> Void) -> Timer {
+        if #available(iOS 10.0, *) {
+            return .scheduledTimer(withTimeInterval: withTimeInterval, repeats: true,
+                                                  block: block)
+        } else {
+            return .iOS9ScheduledTimer(withTimeInterval: withTimeInterval, repeats: true, block: block)
+        }
+    }
+
+    @objc fileprivate static func timerBlcokInvoke(timer: Timer) {
```
Comment (mikeger) [inline Wire-iOS/Sources/Helpers/Timer+allVersionCompatibleScheduledTimer.swift]: Typo `timerBlcokInvoke`

File: Wire-iOS/Sources/Helpers/Timer+allVersionCompatibleScheduledTimer.swift [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+
+import Foundation
+
+
+fileprivate class Block<T> {
```
Comment (mikeger) [inline Wire-iOS/Sources/Helpers/Timer+allVersionCompatibleScheduledTimer.swift]: I would suggest to name this class as `Box` or something similar, as it's seems not to be bound to the block type (and it should not!).

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+BurstTimestamp.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import Foundation
+
+public extension ConversationCell {
+
+    @objc func scheduledTimerForUpdateBurstTimestamp() {
+        guard let _ = layoutProperties else {return}
+
+        if layoutProperties.showBurstTimestamp {
+
+            burstTimestampTimer = .allVersionCompatibleScheduledTimer(withTimeInterval: 60, repeats: true) { [weak self] _ in                                                                        self?.updateBurstTimestamp()
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+BurstTimestamp.swift]: This timer is not doing anything, it is looking suspicious.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Private.h [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+#ifndef ConversationCell_private_h
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+Private.h]: This thing is not necessary in Objective-C, as with `#import` it is not possible to include the same header several times.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m [hunk 5]
```diff
@@ -132,14 +126,17 @@ - (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSStr
     return self;
 }
 
+- (void)dealloc {
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell.m]: Minor: we put the bracket on the next line usually in Obj-C for the reasons of tradition, just a remark.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+BurstTimestamp.swift (hunk 3 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+BurstTimestamp.swift]: burstTimestampTimer invokes updateBurstTimestamp for every 60 sec.

Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+BurstTimestamp.swift]: I see it now, it is off screen due to formatting, please consider putting it on the next line 👍

Comment (mikeger) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+BurstTimestamp.swift]: How I see it: 
screen shot 2017-12-28 at 11 00 29

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConversationCell+BurstTimestamp.swift]: Rewrite the if check to guard, the reduce one indent level.

**Pull request merged**